maxim/logger/langchain/utils.py

```
@@ -400,6 +400,40 @@ def parse_token_usage_for_result(result: LLMResult):
+        # Adding index to each choice
+        for i, choice in enumerate(choices):
+            choices[i] = {**choice, "index": i}
+        return {
+            "id": str(uuid4()),
+            "created": int(time.time()),
+            "choices": choices,
+            "usage": usage,
+        }
+    except Exception as e:
+        logger.error(f"Error parsing LLM result: {str(e)}")
+        raise Exception(f"Error parsing LLM result: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)